### PR TITLE
cli: colocate stories and components, centralize main.js

### PR DESCRIPTION
**Pull request by tooppaaa**
Issue: #10901
https://github.com/storybookjs/storybook/pull/10913 has too much commits / noise

## What I did

Move components from @storybook/< framework >/demo to cli/frameworks/react/js and /ts
Copy stories folder to user src (if any) or to root
mutualise generator

## How to test

E2E should be green
use CLI

### CODE REVIEW

File: lib/cli/src/generators/Generator.ts [hunk 1]
```diff
@@ -1,13 +1,89 @@
...
+
+  const packages = [
+    `@storybook/${framework}`,
+    ...addons,
+    ...extraPackages,
+    ...extraAddons,
+    ...addonsPeerDeps,
+  ].filter(Boolean);
+  const versionedPackages = await packageManager.getVersionedPackages(...packages);
+  console.log('versionedPackages', versionedPackages);
```
Comment (gaetanmaisse) [inline lib/cli/src/generators/Generator.ts]: To remove?

File: lib/cli/versions.json [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-  "@storybook/mithril": "6.0.0-beta.23",
-  "@storybook/preact": "6.0.0-beta.23",
-  "@storybook/rax": "6.0.0-beta.23",
-  "@storybook/react": "6.0.0-beta.23",
-  "@storybook/riot": "6.0.0-beta.23",
-  "@storybook/server": "6.0.0-beta.23",
-  "@storybook/svelte": "6.0.0-beta.23",
-  "@storybook/vue": "6.0.0-beta.23",
-  "@storybook/web-components": "6.0.0-beta.23"
+  "@storybook/angular": "6.0.0-beta.25",
```
Comment (gaetanmaisse) [inline lib/cli/versions.json]: @shilman I think these versions should be bumped automatically with the release no? However, I'm not sure they are used for now.

**Review by gaetanmaisse — APPROVED** [review]
🎉 👏 Awesome work @tooppaaa, I agree with you: 🟢 CI at 5 pm on a Friday, you deserve your weekend 😛

**Review by shilman — APPROVED** [review]
You are amazing @tooppaaa !!! 👏 The changes look fantastic, and congratulations for finally getting it green.

I want to discuss how best to merge this with the work @tmeasday and @domyen have been doing on on the initial project templates. I suspect the answer is to merge this to `next`, release, and then add those changes on top of this, but I want to check first.

**Comment by tooppaaa** [comment]
Absolutely fine. The only change here in addition of the refactoring is collocating components from stories.
I was about to start implementing something from @domyen prototype but I can start with adding essentials !

**Comment by tmeasday** [comment]
@tooppaaa you could definitely start on @domyen's prototype but it might make sense to give it a week or so as we've also been working in the corresponding introductory documentation which makes heavy use of the template so could lead to some tweaks. (If you are interested we can also share the google doc).

**Comment by tooppaaa** [comment]
There are multiple ways for me to setup essentials / stories. It's best to wait you're done with it. This will be a easy but fastidious work, best to do it once.
Definitely interested by the doc to see if there's anything to start with ;)

**Review by ndelangen — APPROVED** [review]

**Pull request merged**